src/Mobs/Path.cpp

```diff
@@ -543,6 +595,91 @@ bool cPath::BodyFitsIn(const Vector3i & a_Location)
 
 
 
+bool cPath::BlockTypeIsSpecial(BLOCKTYPE a_Type)
+{
+	if (BlockTypeIsFence(a_Type))
+	{
+		return true;
+	}
+
+	switch (a_Type)
+	{
+		case E_BLOCK_OAK_DOOR:
+		case E_BLOCK_DARK_OAK_DOOR:
+		case E_BLOCK_TRAPDOOR:
+		case E_BLOCK_WATER:
+		case E_BLOCK_STATIONARY_WATER:
+		{
+			return true;
+		}
+		default:
+		{
+			return false;
+		}
+	}
+}
+
+bool cPath::BlockTypeIsFence(BLOCKTYPE a_Type)
```

Comment:
This should eventually be moved out to cBlockInfo, but I kept it here for now because things might change.